src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -12,14 +11,14 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsManager
     {
-        public static readonly SNIMarsManager Singleton = new SNIMarsManager();
-        private ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections = new ConcurrentDictionary<SNIHandle, SNIMarsConnection>();
+        private readonly ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections;
 
         /// <summary>
         /// Constructor
         /// </summary>
         public SNIMarsManager()
```

Comment:
The MarsManager only needs to be created once _per physical connection_, not once for the whole SqlClient namespace. We create the MarsManager here: https://github.com/dotnet/corefx/pull/22709/files#diff-6c0b5063f5adf829632976a302d48309R24 and then copy it elsewhere here: https://github.com/dotnet/corefx/pull/22709/files#diff-6c0b5063f5adf829632976a302d48309R55